controllers/flinkcluster_submit_job_script.go

```diff
@@ -31,78 +31,101 @@ var submitJobScript = `
+    echo_log "Aborted to submit job." "job_check_log"
+
+    write_term_log "Aborted job submit because JobManager is unavailable." "job_check_log"
+
+    return 1
 }
 
-function check_existing_jobs() {
-	echo "Checking existing jobs..."
-	list_jobs
-	if list_jobs | grep -e "(SCHEDULED)" -e "(CREATED)" -e "(SUSPENDED)" -e "(FINISHED)" -e "(FAILED)" -e "(CANCELED)" \
-		-e "(RUNNING)" -e "(RESTARTING)" -e "(CANCELLING)" -e "(FAILING)" -e "(RECONCILING)"; then
-		echo "Found an existing job, skip resubmitting..."
-		return 0
-	fi
-	return 1
+function submit_job() {
+    local job_id
+
+    # Submit job and extract the job ID
+    echo "/opt/flink/bin/flink run $*" | tee -a submit_log
+    if /opt/flink/bin/flink run "$@" 2>&1 | tee -a submit_log; then
+        local -r job_id_indicator="Job has been submitted with JobID"
+        job_id=$(grep "${job_id_indicator}" submit_log | awk -F "${job_id_indicator}" '{printf $2}' | awk '{printf $1}')
+    fi
+
+    # Write result as YAML format to pod termination-log.
+    # On failure, write log only.
+    if [[ -z ${job_id+x} ]]; then
+        write_term_log "Failed to submit Flink job." "submit_log"
+        return 1
+    fi
+
+    # On success, write job ID and log.
+    echo "jobID: ${job_id}" >"${TERM_LOG}"
+    write_term_log "Successfully Flink job submitted!" "submit_log"
+
+    return 0
 }
 
-function submit_job() {
-	echo -e "\nSubmitting job..."
-	echo "/opt/flink/bin/flink run $@"
-	/opt/flink/bin/flink run "$@"
+function echo_log() {
+    local msg="$1"
+    local log_file="$2"
+    echo -e "${msg}" | tee -a "${log_file}"
 }
 
-function wait_for_job() {
-	while true; do
-		echo -e "\nWaiting for job to finish..."
-		list_jobs
-
-		# Find active job first.
-		# If the current job is restarted by the operator, there will be records of past stopped jobs.
-		# TODO: It needs to be improved to determine the job state with the submitted job id.
-		if list_jobs | grep -e "(SCHEDULED)" -e "(CREATED)" -e "(SUSPENDED)" \
-			-e "(RUNNING)" -e "(RESTARTING)" -e "(CANCELLING)" -e "(FAILING)" -e "(RECONCILING)"; then
-			echo -e "\nFound an active job."
-		else
-			if list_jobs | grep "(FINISHED)"; then
-				echo -e "\nJob has completed successfully, exiting 0"
-				return 0
-			fi
-			if list_jobs | grep "(FAILED)"; then
-				echo -e "\nJob failed, exiting 1"
-				return 1
-			fi
-			if list_jobs | grep "(CANCELED)"; then
-				echo -e "\nJob has been cancelled, exiting 2"
-				return 2
-			fi
-			if list_jobs | grep "No running jobs" && list_jobs | grep "No scheduled jobs"; then
-				echo -e "\nNo running and scheduled jobs, exiting 3"
-				return 3
-			fi
-			echo -e "\nUnknown job state, check it again in the next iteration."
-		fi
-		sleep 30
-	done
+function write_term_log() {
+    local msg="$1"
+    local log_file="$2"
+
+    echo "message: |" >>"${TERM_LOG}"
+    echo "  ${msg}" >>"${TERM_LOG}"
+
+    IFS=''
+    while read -r line; do
```

Comment:
I don't quite understand what you are trying to do here, could you add a comment?

Reply:
Commented.